test/solidlsp/ocaml/test_ocaml_basic.py

```diff
@@ -0,0 +1,183 @@
+        # Line 8: "let rec fib n =" - 'fib' starts at character 8 (0-indexed)
+        fib_line = 7  # 0-indexed line number
+        fib_char = 8  # 0-indexed character position
+
+        refs = language_server.request_references(file_path, fib_line, fib_char)
+
+        # Should find at least 3 references: definition + 2 recursive calls in same file
+        assert len(refs) >= 3, f"Expected at least 3 references to fib (definition + 2 recursive), found {len(refs)}"
+
+        # All references should be in lib/test_repo.ml (same file as definition)
+        # Use forward slashes for URI matching (URIs always use /)
+        lib_refs = [ref for ref in refs if "lib/test_repo.ml" in ref.get("uri", "")]
+        assert len(lib_refs) >= 3, f"Expected at least 3 references in lib/test_repo.ml, found {len(lib_refs)}"
+
+    @pytest.mark.parametrize("language_server", [Language.OCAML], indirect=True)
+    def test_mixed_ocaml_modules(self, language_server: SolidLanguageServer) -> None:
+        """Test that the language server can find symbols from OCaml modules"""
+        # Test that full symbol tree includes symbols from various file types
+        all_symbols = language_server.request_full_symbol_tree()
+
+        # Should find symbols from main OCaml files
+        assert SymbolUtils.symbol_tree_contains_name(all_symbols, "fib"), "Should find fib from .ml file"
+        assert SymbolUtils.symbol_tree_contains_name(all_symbols, "DemoModule"), "Should find DemoModule from .ml file"
+        assert SymbolUtils.symbol_tree_contains_name(all_symbols, "someFunction"), "Should find someFunction from DemoModule"
+        assert SymbolUtils.symbol_tree_contains_name(all_symbols, "num_domains"), "Should find num_domains constant"
+
+    def test_reason_file_patterns(self) -> None:
+        """Test that OCaml language configuration recognizes Reason file extensions"""
+        from solidlsp.ls_config import Language
+
+        ocaml_lang = Language.OCAML
+        file_matcher = ocaml_lang.get_source_fn_matcher()
+
+        # Test OCaml extensions
+        assert file_matcher.is_relevant_filename("test.ml"), "Should match .ml files"
+        assert file_matcher.is_relevant_filename("test.mli"), "Should match .mli files"
+
+        # Test Reason extensions
+        assert file_matcher.is_relevant_filename("test.re"), "Should match .re files"
+        assert file_matcher.is_relevant_filename("test.rei"), "Should match .rei files"
+
+        # Test non-matching extensions
+        assert not file_matcher.is_relevant_filename("test.py"), "Should not match .py files"
+        assert not file_matcher.is_relevant_filename("test.js"), "Should not match .js files"
+
+    @pytest.mark.parametrize("language_server", [Language.OCAML], indirect=True)
+    def test_find_references_across_files(self, language_server: SolidLanguageServer) -> None:
+        """Test finding references across .ml files in lib, bin, and test directories.
```

Comment:
this duplicates test_cross_file_refs.py, right? Pls either remove the other one or this one, keep whichever is more useful